File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/proxy/simple/Actual.java
// Actual.java

package com.bigboxcode.designpattern.proxy.simple;

public class Actual implements ProxyInterface {

    @Override
    public void operation1() {
        System.out.println("Performing operation 1 in the Actual Object");
    }

    @Override
    public void operation2() {
        System.out.println("Performing operation 2 in the Actual Object");
    }

}

File: design-patterns/go/proxy/simple/actual.go
// actual.go

package main

import "fmt"

type Actual struct {}

func NewActual() (actual *Actual) {
	actual = &Actual{}
	return
}
func (actual *Actual) Operation1() {
	fmt.Println("Performing operation 1 in the Actual Object")
}

func (actual *Actual) Operation2() {
	fmt.Println("Performing operation 2 in the Actual Object")
}